### PR TITLE
Add term new

### PR DESCRIPTION
**Pull request by daniellanyi**
### Description

Adding markdown file for the __new__() term in Python Dunder Methods

### Type of Change

- Adding a new entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/python/concepts/dunder-methods/terms/new/new.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+---
+Title: '__new__()'
+Description: 'Static method which creates a new instance of a class'
```
Comment (SSwiniarski) [inline content/python/concepts/dunder-methods/terms/new/new.md:3]:
```suggestion
Description: 'Static method which creates a new instance of a class.'
```

File: content/python/concepts/dunder-methods/terms/new/new.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  - 'Classes'
+  - 'Methods'
+  - 'Objects'
+  - 'OOP'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+The **`__new__()`** method creates a new instance of a class cls and takes in the class (of which the instance was requested) as the first argument.
```
Comment (SSwiniarski) [inline content/python/concepts/dunder-methods/terms/new/new.md:19]:
```suggestion
The **`__new__()`** method creates a new instance of a class `cls` and takes in the class (of which the instance was requested) as the first argument.
```

File: content/python/concepts/dunder-methods/terms/new/new.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+## Syntax
+
+```pseudo
+class ClassName:
+  __new__(cls, *args, **kwargs):
+    obj = super().__new__(cls)
+    return obj
+```
+
+Object creation is typically done by invoking the superclass’ `__new__()` in two ways. The first argument of `__new__()` is always the class which is passed in as the first parameter automatically. The name cls is not a keyword; it’s used to reference the class (the first parameter) by convention. When `__new__()` returns an instance of cls it automatically invokes the [`__init__()`](https://www.codecademy.com/resources/docs/python/dunder-methods/init) method of that instance with the arguments passed to it. The first argument passed to `__init__()` will be the instance itself (this happens automatically) which is by convention referenced by self.
```
Comment (SSwiniarski) [inline content/python/concepts/dunder-methods/terms/new/new.md:30]:
```suggestion
Object creation is typically done by invoking the superclass’ `__new__()` in two ways. The first argument of `__new__()` is always the class which is passed in as the first parameter automatically. The name `cls` is not a keyword; it’s used to reference the class (the first parameter) by convention. When `__new__()` returns an instance of `cls` it automatically invokes the [`__init__()`](https://www.codecademy.com/resources/docs/python/dunder-methods/init) method of that instance with the arguments passed to it. The first argument passed to `__init__()` will be the instance itself (this happens automatically) which is by convention referenced by `self`.
```

File: content/python/concepts/dunder-methods/terms/new/new.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+Let's add the following to the code above:
+
+```pseudo
+def __init__(self, *args, **kwargs):
+    print(*args, **kwargs)
+
+class_instance = ClassName(1, 2)
+```
+First `__new__()` is invoked during the construction of the object (instance of class ClassName) with the arguments 1 and 2 which are accepted within \*args. The object is created by calling the `__new__()` method of the object base class which only takes in the class cls. It then returns an object which is then stored in obj and returned. This invokes the `__init__()` method of the ClassName class with the arguments (obj, \*args, \*\*kwargs).
```
Comment (SSwiniarski) [inline content/python/concepts/dunder-methods/terms/new/new.md:42]:
```suggestion

First `__new__()` is invoked during the construction of the object (instance of class `ClassName`) with the arguments `1` and `2` which are accepted within `\*args`. The object is created by calling the `__new__()` method of the object base class which only takes in the class `cls`. It then returns an object which is then stored in `obj` and returned. This invokes the `__init__()` method of the `ClassName` class with the arguments `(obj, \*args, \*\*kwargs)`.
```

File: content/python/concepts/dunder-methods/terms/new/new.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+This method has two main applications:
+
+- Customizing how new class instances are created.
+- Subclassing of immutable types which cannot be initialized via a simple __init__method.
```
Comment (SSwiniarski) [inline content/python/concepts/dunder-methods/terms/new/new.md:47]:
```suggestion
- Subclassing of immutable types which cannot be initialized via a simple `__init__()` method.
```

File: content/python/concepts/dunder-methods/terms/new/new.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+
+## Codebyte Example 1
```
Comment (SSwiniarski) [inline content/python/concepts/dunder-methods/terms/new/new.md:50]: We generally only have one codebyte example at the end of the entry.  We can make this just a regular example with a ```py block.

File: content/python/concepts/dunder-methods/terms/new/new.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+class MyInt(int):
+    def __new__(cls, value, bitLength = None):
+        obj = int.__new__(cls, value)
+        obj.bitLength = bitLength
+        return obj
+
+int1 = MyInt(2, 4)
+print(int1, int1.bitLength)
+```
+[Official Python documentation](https://docs.python.org/3/reference/datamodel.html#object.__new__)
```
Comment (SSwiniarski) [inline content/python/concepts/dunder-methods/terms/new/new.md:86]: Remove this last link.

File: content/python/concepts/dunder-methods/terms/new/new.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        obj = super().__new__(cls)
+        return obj
+
+    def __init__(self, value):
+        self.value = value
+```
+
+## Codebyte Example 2
+
+
```
Comment (SSwiniarski) [inline content/python/concepts/dunder-methods/terms/new/new.md:72]:
```suggestion
```
remove extra line break.

File: content/python/concepts/dunder-methods/terms/new/new.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+Creating a subclass of the int class called MyInt. MyInt will have a mutable bitLength property. Note that when the int class’ `__new__()` method is invoked `value` is passed in as the second argument. When `__new__()` returns an int object it will initialise it with `value`. The bitlength attribute as well as any other attribute can also be initialised within the `__new__()` method of myInt.
+
+
```
Comment (SSwiniarski) [inline content/python/concepts/dunder-methods/terms/new/new.md:75]:
```suggestion
```
Remove extra line break

File: content/python/concepts/dunder-methods/terms/new/new.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+class_instance = ClassName(1, 2)
+```
+First `__new__()` is invoked during the construction of the object (instance of class ClassName) with the arguments 1 and 2 which are accepted within \*args. The object is created by calling the `__new__()` method of the object base class which only takes in the class cls. It then returns an object which is then stored in obj and returned. This invokes the `__init__()` method of the ClassName class with the arguments (obj, \*args, \*\*kwargs).
+
+This method has two main applications:
+
+- Customizing how new class instances are created.
+- Subclassing of immutable types which cannot be initialized via a simple __init__method.
+
+
```
Comment (SSwiniarski) [inline content/python/concepts/dunder-methods/terms/new/new.md:49]:
```suggestion
```
remove extra line break

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@daniellanyi I did a first review and have a first round of comments.

**Comment by SSwiniarski** [comment]
@daniellanyi, It's been a while, do you need assistance implementing the suggestions above?

**Comment by daniellanyi** [comment]
Thank you!. I have applied the first round of changes you've suggested to
clean up the markdown. I'm sorry I didn't reply sooner; there is a time
zone difference (I live in the UK). I hope I've worded the description of
the '__new__()' method accurately and clearly. If you have more suggestions
on how to improve it feel free to let me know.

**Comment by SSwiniarski** [comment]
@daniellanyi, you still have two codebyte blocks. I'd like you to turn the first one into a regular example with a ```py block.

**Comment by SSwiniarski** [comment]
@daniellanyi don't forget to change the headers on the last two sections to reflect the change of the codebyte block. ie: change from "Codebyte Example 1" and "Codebyte Example 2" to "Example" and "Codebyte Example"

**Review by SSwiniarski — APPROVED** [review]
Looks good for review two.

File: content/python/concepts/dunder-methods/terms/new/new.md [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+```pseudo
+class ClassName:
+  __new__(cls, *args, **kwargs):
+    obj = super().__new__(cls)
+    return obj
+```
+
+Object creation is typically done by invoking the superclass’ `__new__()` in two ways. The first argument of `__new__()` is always the class which is passed in as the first parameter automatically. The name `cls` is not a keyword; it’s used to reference the class (the first parameter) by convention. When `__new__()` returns an instance of `cls` it automatically invokes the [`__init__()`](https://www.codecademy.com/resources/docs/python/dunder-methods/init) method of that instance with the arguments passed to it. The first argument passed to `__init__()` will be the instance itself (this happens automatically) which is by convention referenced by `self`.
+
+In the example above no parent class was specified therefore the class will directly inherit from the object base class (from which all classes inherit). Therefore calling it explicitly using `object.__new__(cls)` will have the same effect.
```
Comment (KTom101) [inline content/python/concepts/dunder-methods/terms/new/new.md:32]:
```suggestion
In the example above, no parent class was specified, therefore, the class will directly inherit from the object base class (from which all classes inherit). Calling it explicitly using `object.__new__(cls)` will have the same effect.
```

File: content/python/concepts/dunder-methods/terms/new/new.md [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+First `__new__()` is invoked during the construction of the object (instance of class `ClassName`) with the arguments `1` and `2` which are accepted within `\*args`. The object is created by calling the `__new__()` method of the object base class which only takes in the class `cls`. It then returns an object which is then stored in `obj` and returned. This invokes the `__init__()` method of the `ClassName` class with the arguments `(obj, \*args, \*\*kwargs)`.
+
+This method has two main applications:
+
+- Customizing how new class instances are created.
+- Subclassing of immutable types which cannot be initialized via a simple `__init__()` method.
+
+## Example
+
+In this example the number of instances of Test is limited to ‘MaxNumberOfInstances’:
```
Comment (KTom101) [inline content/python/concepts/dunder-methods/terms/new/new.md:52]:
```suggestion
In the following example, the number of instances of the class `Test` is limited to ‘MaxNumberOfInstances’:
```

**Review by KTom101 — CHANGES_REQUESTED** [review]
@daniellanyi Second review is complete!

**Review by KTom101 — APPROVED** [review]
@daniellanyi I made some small changes [6104f5f](https://github.com/Codecademy/docs/pull/1573/commits/6104f5fa247fc09df3ed050faddec8379653bc97) & [7ab2cfb](https://github.com/Codecademy/docs/pull/1573/commits/7ab2cfb8f62ea1881ba3d6df9e8207619f5d3ba9), otherwise looks good to me! 👍

**Comment by KTom101** [comment]
@daniellanyi Entry is live here: https://www.codecademy.com/resources/docs/python/dunder-methods/new

You're currently listed as an anonymous contributor:

Screenshot 2023-03-08 at 3 10 28 AM

To be credited, follow these steps below:

1.) Add your GitHub Username to your Codecademy Profile Details, you can get there on your desktop by pressing on your profile picture in the upper right-hand corner and selecting "Profile" in the drop-down box:

YourGitHubName

2.) Then log out and log back into Codecademy using your GitHub login:

ReLogIn

> Note:  Your email used for Codecademy will have to be listed in your GitHub profile for the login feature to properly work. If you use the same email for both GitHub and Codecademy already, then you're set.

Let me know if you need any help if you decide to do so! 😁

**Pull request merged**